### ISSUE TITLE
intermittent failure to find passwords (possible race?)

### ISSUE THREAD

**Issue by sakoht** [abbot/go-http-auth#66]
I'm finding this works great on most of my sites, and has been a drop-in replacement for Apache, but I have some exceptions.  On a page that embeds two iframes that hit my protected (ie hits the server to check passwords in parallel twice at nearly the same moment), even if I have the password cached already, it fails to find a password, and gets int a loop where no matter how many times I enter it, it fails.

Curious if there is a race condition behind this.  I'm attempting to debug.

**Comment by abbot** [comment]
Can you suggest a minimal setup (Go code + HTML 'frontend')  to reproduce the issue?

**Comment by sakoht** [comment]
Thanks, and sorry for the slow response.  

I've put the full source of _all_ of my Go code in this gist.
The few lines that were added to incorporate http auth are highlighted:
https://gist.github.com/sakoht/9b5679ed3d2f8337b719c550c5fef33b#file-reverseproxy-go-L89-L93

I attempted to pull out a full repo to demo the problem, but that proved complicated.  The golang app reverse-proxies to several in-house services, and the simple routes work fine.  The first ones that produce an error includes _two_ iframes on a dummy client site where the iframes are both protected by this software, and are, of course, hit at virtually the same moment by the browser, which is why I suspected a race.

If you notice anything foolish in my code above please let me know.  I'm going to attempt to debug with `log.Printf` added to your `digest.go` so I can see how/where it is failing.  If I can fix it and send a patch, I'll make a pull request.

**Comment by sakoht** [comment]
(Note: `Check` and `Wrap` in my source file are copied from your library so I could add some debug code.)

**Comment by sakoht** [comment]
This was not a race.  The two iframes both eventually use AJAX to talk to a Django REST Framework back-end that goes through this Go reverse-proxy.  DRF sets the `Authentication` header to `Token XXXXXX`, which overwrites the header made by this library.

This worked previously with apache2, so I am guessing the issue is that there were previously multiple `Authentication` headers and apache2 ignored the one it didn't need, and Django did the same.

I'll make a separate issue and close this one.  And hopefully have a PR for you...
